tests/OAuth2SummitOrderExtraQuestionTypeApiTest.php

```diff
@@ -716,4 +767,282 @@ public function testGetAllMainBySummit(){
+
+            $sq->addAllowedTicketType($attendee_ticket_type);
+
+            self::$questions[] = $sq;
+            self::$summit->addOrderExtraQuestion($sq);
+
+            $rule1 = new SubQuestionRule();
+            $rule1->setAnswerValues([self::$values[0]->getId()]);
+            $rule1->setVisibility(ExtraQuestionTypeConstants::SubQuestionRuleVisibility_Visible);
+            $rule1->setVisibilityCondition(ExtraQuestionTypeConstants::SubQuestionRuleVisibilityCondition_Equal);
+            $rule1->setAnswerValuesOperator(ExtraQuestionTypeConstants::SubQuestionRuleAnswerValuesOperator_Or);
+
+            $question1->addSubQuestionRule($rule1);
+            $sq->addParentRule($rule1);
+        }
+
+        // Subquestion without allowed ticket type
+
+        $no_ticket_type_sq = new SummitOrderExtraQuestionType();
+        $no_ticket_type_sq->setUsage(SummitOrderExtraQuestionTypeConstants::BothQuestionUsage);
+        $no_ticket_type_sq->setLabel('Country');
+        $no_ticket_type_sq->setName('Country');
+        $no_ticket_type_sq->setType(ExtraQuestionTypeConstants::TextQuestionType);
+        $no_ticket_type_sq->setPrintable(true);
+        $no_ticket_type_sq->setMandatory(true);
+
+        self::$questions[] = $no_ticket_type_sq;
+        self::$summit->addOrderExtraQuestion($no_ticket_type_sq);
+
+        $rule1 = new SubQuestionRule();
+        $rule1->setAnswerValues([self::$values[0]->getId()]);
+        $rule1->setVisibility(ExtraQuestionTypeConstants::SubQuestionRuleVisibility_Visible);
+        $rule1->setVisibilityCondition(ExtraQuestionTypeConstants::SubQuestionRuleVisibilityCondition_Equal);
+        $rule1->setAnswerValuesOperator(ExtraQuestionTypeConstants::SubQuestionRuleAnswerValuesOperator_Or);
+
+        $question1->addSubQuestionRule($rule1);
+        $no_ticket_type_sq->addParentRule($rule1);
+
+        self::$em->persist(self::$summit);
+        self::$em->flush();
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+         $params = [
+            'id' => self::$summit->getId(),
+            'attendee_id' => $attendee->getId(),
+            'page'     => 1,
+            'per_page' => 10,
+            'order'    => '+id',
+            // recursive expand
+            'expand' => '*sub_question_rules,*sub_question,values,values.question'
+        ];
+
+        $response = $this->action(
+            "GET",
+            "OAuth2SummitOrderExtraQuestionTypeApiController@getAttendeeExtraQuestions",
+            $params,
+            [],
+            [],
+            [],
+            $headers
+        );
+
+        $content = $response->getContent();
+        $questions = json_decode($content);
+        $this->assertTrue(!is_null($questions));
+        $this->assertEquals($questions->total, count($allowed_subquestion_labels) + 1); //allowed subquestions + allowed main question
```

Comment:
this test result is not correct
pleas check response of https://github.com/fntechgit/fnEvent/blob/88414da1ef3960b41b2630bcca3bb12d112ab434/src/actions/summit-actions.js#L32
should only return the main questions ( roots ) so in this case is 2

Reply:
done